### PR TITLE
Goggles - Add individual files for config classes

### PR DESCRIPTION
**Pull request by DartRuffian**
**When merged this pull request will:**
- Move root classes from `config.cpp` to each of their own `*.hpp` files..
- Move macros to a `script_macros.hpp` file.
  - Some of these might make more sense in a `defines.hpp`
- Add `IDD` / `IDC` macros for dialogs and controls.

*No functionality was changed, just organization.*

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by DartRuffian** [comment]
I got annoyed when looking at things in goggles, so I wanted to PR moving stuff to files

File: addons/goggles/script_macros.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#define COMBAT_GOGGLES         ACE_Overlay = QPATHTOF(textures\HUD\CombatGoggles.paa); \
+        ACE_OverlayCracked = QPATHTOF(textures\HUD\CombatGogglesCracked.paa); \
+        ACE_Resistance = 2; \
+        ACE_Protection = 1;
```
Comment (Mike-MF) [inline addons/goggles/script_macros.hpp:4]: I know it's just a copy/paste, but this bothers me. Will require all uses of `COMBAT_GOGGLES` to now have a semicolon.

```suggestion
#define COMBAT_GOGGLES \
	ACE_Overlay = QPATHTOF(textures\HUD\CombatGoggles.paa); \
        ACE_OverlayCracked = QPATHTOF(textures\HUD\CombatGogglesCracked.paa); \
        ACE_Resistance = 2; \
        ACE_Protection = 1
```

Comment (DartRuffian) [reply addons/goggles/script_macros.hpp:4]: Fine by me, only reason I didn't change it was because I didn't want to change the actual definitions.

**Comment by DartRuffian** [comment]
Should the goggles changes in `compat_spe` be moved to its own addon and include the `script_macros.hpp`?

File: addons/goggles/script_macros.hpp (hunk 1 above)
Comment (Mike-MF) [reply addons/goggles/script_macros.hpp]: You didn't make the full change, just the semicolon.

Comment (DartRuffian) [reply addons/goggles/script_macros.hpp]: I didn't notice it, but the extra line feels unnecessary

Comment (Mike-MF) [reply addons/goggles/script_macros.hpp]: it's also still indented too far. You should have just committed the suggestion.

Comment (DartRuffian) [reply addons/goggles/script_macros.hpp]: It didn't let me when you first made it

**Review by Mike-MF — APPROVED** [review]

**Comment by Mike-MF** [comment]
> Should the goggles changes in `compat_spe` be moved to its own addon and include the `goggles\script_macros.hpp`?

Forgot to respond to this, It was said somewhere, either on here or Discord that turning goggles into subaddons wasn't desired.

**Review by PabstMirror — APPROVED** [review]

**Review by johnb432 — APPROVED** [review]

**Pull request merged**